### PR TITLE
Add UUIDs to test template names and test bucket names

### LINKED ISSUE #4996: Refactor Media tests to use uuids
500 errors are blocking #4980 and #4994 - they're caused by concurrent job template create requests with the same name - UUIDs should fix that (and if not ping me and we can talk about other options)

cc @irataxy

### PR DESCRIPTION
**Pull request by irataxy**
## Description

Fixes #4996

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [X ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [X] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [X] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Comment by busunkim96** [comment]
@irataxy I think `snippet-bot` is having some trouble today, please disregard the message above.

File: media/transcoder/job_test.py [hunk 1]
```diff
@@ -51,14 +52,17 @@ def test_job_operations(capsys):
     # Enable the following API on the test project:
     # *   Transcoder API
 
-    _clean_bucket()
+    _delete_bucket()
```
Comment (busunkim96) [inline media/transcoder/job_test.py:55]: Could you tweak these tests to use pytest fixtures instead? This increases the likelihood resources are cleaned up even if the test happens to fail midway.

* https://github.com/GoogleCloudPlatform/python-docs-samples/blob/a777e9d760bcf6b678474f930d9e9a329ddcd30f/storage/cloud-client/snippets_test.py#L96-L105
* https://github.com/GoogleCloudPlatform/python-docs-samples/blob/a777e9d760bcf6b678474f930d9e9a329ddcd30f/storage/cloud-client/snippets_test.py#L126-L132

**Review by busunkim96 — CHANGES_REQUESTED** [review]

File: media/transcoder/job_test.py (hunk 1 above)
Comment (irataxy) [reply media/transcoder/job_test.py]: Done

**Review by leahecole — APPROVED** [review]
LGTM if you're happy with the changes @busunkim96

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**